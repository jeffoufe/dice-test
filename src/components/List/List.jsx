// @flow

import React from 'react';
import styles from './List.module.scss';
import { BlueHeader, ImportantText, Text, GreyHeader } from '../../components/Typography/Typography';

type ListEntryProps = {
    primary: string,
    secondary?: string,
    tertiary?: boolean
};

type ListProps = {
    list: Array<ListEntryProps>,
    title: string
}

const List = ({ list, title }: ListProps) => (
    <div>
        <BlueHeader>{title}</BlueHeader>
        {list.map(({ primary, secondary, tertiary }: ListEntryProps, index: number) => (
            <div className={styles['list-entry']} key={`list-entry-${index}`}>
                <Text inline>{primary}</Text>
                {secondary && <ImportantText inline> &#65293; {secondary}</ImportantText>}
                {tertiary && <GreyHeader inline> Sold out</GreyHeader>}
            </div>
        ))}
    </div>
)

export default List;