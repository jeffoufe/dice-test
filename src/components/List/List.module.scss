.list-entry:not(:first-child) {
    margin-top: 20px;
}