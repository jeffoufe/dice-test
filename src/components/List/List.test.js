import List from './List';
import { shallow } from 'enzyme';
import React from 'react';

describe('List', () => {
    const props = {
        title: 'test',
        list: [
            { primary: 'primary' },
            { primary: 'primary', secondary: 'secondary' },
            { primary: 'primary', secondary: 'secondary', tertiary: 'tertiary' },
        ]
    }

    const wrapper = shallow(
        <List {...props} />
    );

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('conditional rendering', () => {
        expect(wrapper.find('Text').length).toEqual(3);
        expect(wrapper.find('ImportantText').length).toEqual(2);
        expect(wrapper.find('GreyHeader').length).toEqual(1);
    })
})