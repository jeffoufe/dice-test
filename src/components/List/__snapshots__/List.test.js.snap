// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <BlueHeader>
    test
  </BlueHeader>
  <div
    className="list-entry"
    key="list-entry-0"
  >
    <Text
      inline={true}
    >
      primary
    </Text>
  </div>
  <div
    className="list-entry"
    key="list-entry-1"
  >
    <Text
      inline={true}
    >
      primary
    </Text>
    <ImportantText
      inline={true}
    >
       － 
      secondary
    </ImportantText>
  </div>
  <div
    className="list-entry"
    key="list-entry-2"
  >
    <Text
      inline={true}
    >
      primary
    </Text>
    <ImportantText
      inline={true}
    >
       － 
      secondary
    </ImportantText>
    <GreyHeader
      inline={true}
    >
       Sold out
    </GreyHeader>
  </div>
</div>
`;
